source/simplex/fe_lib.cc

```diff
@@ -151,6 +151,78 @@ namespace Simplex
       return unit_face_points;
     }
 
+    /**
+     * Specify the constraints which the dofs on the two sides of a cell
+     * interface underlie if the line connects two cells of which one is refined
+     * once.
+     */
+    template <int dim>
+    FullMatrix<double>
+    constraints_fe_poly(const unsigned int /*degree*/)
+    {
+      // no constraints in 1d
+      // constraints in 3d not implemented yet
+      return FullMatrix<double>();
+    }
+
+    template <>
+    FullMatrix<double>
+    constraints_fe_poly<2>(const unsigned int degree)
+    {
+      const unsigned int dim = 2;
+
+      Assert(degree <= 2, ExcNotImplemented());
+
+      // the following implements the 2d case
+      // (the 3d case is not implemented yet)
+      //
+      // consult FE_Q_Base::Implementation::initialize_constraints()
+      // for more information
+
+      std::vector<Point<dim - 1>> constraint_points;
+      // midpoint
+      constraint_points.emplace_back(0.5);
+      if (degree == 2)
+        {
+          // midpoint on subface 0
+          constraint_points.emplace_back(0.25);
+          // midpoint on subface 1
+          constraint_points.emplace_back(0.75);
+        }
+
+      // Now construct relation between destination (child) and source (mother)
+      // dofs.
+
+      const unsigned int n_dofs_constrained = constraint_points.size();
+      unsigned int       n_dofs_per_face    = degree + 1;
+
+      FullMatrix<double> interface_constraints;
+      {
+        TableIndices<2> interface_constraints_size = {n_dofs_constrained,
+                                                      n_dofs_per_face};
+        interface_constraints.TableBase<2, double>::reinit(
+          interface_constraints_size);
+      }
```

Comment:
What does this block do? Isn't it just setting the size of `interface_constraints`, and that could already be done in the constructor call above the `{...}` block?